### ISSUE TITLE
Version the build task DLLs on every build

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#31081]
When building with multiple versions of Visual Studio previews installed you may end up with non-sense errors like the following: 

> D:\VS\src\Tools\Roslyn\Microsoft.CSharp.Core.targets(91,10): error MSB4064: The "NullableReferenceTypes" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property

This happens when MSBuild ends up loading the same MS.CA.Build.Tasks assemblies for different target files. The build in an ideal state should always load the DLL that is paired with a given targets file. This isn't guaranteed to happen today though. 

For desktop build we can fix this by moving to build scheme where the version number of our build task changes on every build. That would make the DLLs unique and hence won't conflict when MSBuild loads them.

**Comment by AArnott** [comment]
To clarify (and I know @jaredpar already knows this): we need the *assembly version* to be unique for each build.

**Comment by agocke** [comment]
I don't think you mean literally every build. That would break up-to-date. Some kind of calculation based on hash? Bump every time there's a public API break?

**Comment by jaredpar** [comment]
Clarified to say "assembly version" and "official build".

**Comment by jaredpar** [comment]
This is still scheduled for 16.0 but it's not high on our priority list at the moment.

**Comment by AArnott** [comment]
If you were using Nerdbank.GitVersioning in your repo, I'd send a PR with about 3 lines to fix it. :wink:
